packages/react-aria-components/docs/Autocomplete.mdx

```diff
@@ -116,6 +124,452 @@ import {UNSTABLE_Autocomplete as Autocomplete, SearchField, Menu} from 'react-ar
+
+This example shows a `FieldGroup` component that renders a group of filters with a title. The entire group can be given the same `filter` prop, which is passed to all child selects via the `AutocompleteContext` provider.
+
+```tsx example
+import {UNSTABLE_AutocompleteContext as AutocompleteContext, Menu, MenuItem} from 'react-aria-components';
+
+interface FieldGroupProps {
+  title?: string,
+  children?: React.ReactNode
+}
+
+function FieldGroup({title, children}: FieldGroupProps) {
+  let {contains} = useFilter({sensitivity: 'base'});
+  return (
+    <fieldset>
+      <legend>{title}</legend>
+      {/*- begin highlight -*/}
+      <AutocompleteContext.Provider value={{filter: contains}}>
+      {/*- end highlight -*/}
+        {children}
+      </AutocompleteContext.Provider>
+    </fieldset>
+  );
+}
+
+<FieldGroup title="Filters">
+  <MyAutocomplete label="Actions" placeholder="Search actions...">
+    <MyItem id="draft">Draft</MyItem>
+    <MyItem id="published">Published</MyItem>
+    <MyItem id="deleted">Deleted</MyItem>
+  </MyAutocomplete>
+  <MyAutocomplete label="People" placeholder="Search people...">
+    <MyItem id="john">John</MyItem>
+    <MyItem id="emma">Emma</MyItem>
+    <MyItem id="tim">Tim</MyItem>
+  </MyAutocomplete>
+</FieldGroup>
+```
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+fieldset {
+  padding: 1.5em;
+  width: fit-content;
+  display: flex;
+  flex-direction: column;
+  gap: 1em;
+}
+```
+
+</details>
+
+### Custom children
+
+Autocomplete passes props to its child components, such as the label and input, via their associated contexts. These contexts are exported so you can also consume them in your own custom components. This enables you to reuse existing components from your app or component library together with React Aria Components.
+
+<ContextTable components={['TextField', 'SearchField', 'Menu', 'ListBox']} docs={docs} />
+
+This example consumes from `LabelContext` in an existing styled label component to make it compatible with React Aria Components. The <TypeLink links={docs.links} type={docs.exports.useContextProps} /> hook merges the local props and ref with the ones provided via context by Autocomplete.
+
+```tsx
+import type {LabelProps} from 'react-aria-components';
+import {LabelContext, useContextProps} from 'react-aria-components';
+
+const MyCustomLabel = React.forwardRef((props: LabelProps, ref: React.ForwardedRef<HTMLLabelElement>) => {
+  // Merge the local props and ref with the ones provided via context.
+  ///- begin highlight -///
+  [props, ref] = useContextProps(props, ref, LabelContext);
+  ///- end highlight -///
+
+  // ... your existing Label component
+  return <label {...props} ref={ref} />;
+});
+```
+
+Now you can use `MyCustomLabel` within an `Autocomplete`, in place of the builtin React Aria Components `Label`.
+
+```tsx
+<Autocomplete>
+  {/*- begin highlight -*/}
+  <MyCustomLabel>Name</MyCustomLabel>
```

Comment:
Wouldn't this need to be inside a TextField/SearchField? Wonder if this example makes sense either or if you should just follow the example on the field docs pages